src/lib/queries.ts

```
@@ -33,12 +33,25 @@ export type AvailabilityEvent = {
 // -------------------- USER RELATED ----------------------------
 
 export const signUp = async (email: string, name: string): Promise<AuthUser> => {
-  const { data } = await adminSupabase.auth.admin.createUser({
+  let { error } = await adminSupabase.auth.signInWithOtp({
     email,
-    user_metadata: { name },
   });
 
-  return { user_id: data.user!.id, email: data.user!.email! };
+  if (error) {
+    const { data: createUserData, error: createUserError } = await adminSupabase.auth.admin.createUser({
+      email,
+      user_metadata: { name },
+    });
+
+    if (createUserError) {
+      throw createUserError;
+    }
+    return { user_id: createUserData.user.id, email: createUserData.user.email! };
+  }
+
+  const { data } = await supabase.auth.getSession();
+
+  return { user_id: data.session!.user.id, email: data.session!.user.email! };
```

could you check if this exists and throw if not instead of returning a possible null